.github/pull_request_template.md

```diff
@@ -0,0 +1,28 @@
+## Description
+<!-- Provide a clear and concise description of your changes -->
+
+## Type of Change
+<!-- Mark the appropriate option with an [x] -->
+- [ ] Bug fix (non-breaking change that fixes an issue)
+- [ ] New feature (non-breaking change that adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] Documentation update
+- [ ] Refactoring (no functional changes)
+- [ ] Performance improvement
+- [ ] Test updates
+- [ ] CI/CD related changes
+- [ ] Other (please describe):
+
+## Testing
+<!-- Describe the tests you ran and/or added to verify your changes -->
+- [ ] Added new unit tests
+- [ ] Added new integration tests
+- [ ] Ran existing test suite
+- [ ] Tested manually
+
+## Related Issues
+<!-- Link any related issues using the GitHub issue syntax: #issue-number -->
+Fixes #
```

Comment:
should we add a comment saying "if there is no open issue for this, please add details of the problem or open a new issue" ?